tests/shouldwork/Issues/T2046C.hs

```diff
@@ -0,0 +1,21 @@
+module T2046C where
+
+import Clash.Prelude
+import T2046BType
+
+inBit :: (Enum a) => a -> Bool
+inBit x = case fromEnum x of
+        0 -> True
+        1 -> True
+        _ -> False
```

Comment:
Wouldn't GHC optimize this out? I think we might need a `NOINLINE`.

Reply:
No, it doesn't

Reply:
> GHC won't constant fold because the fromEnum it would pick is NOINLINE anyway (since it's a clash prim)
> So the simplifier only gets as far as
> ``` case Clash.Whatever.fromEnum ... of```
> and gives up